examples/js/bots/liquidity_bot.ts

```diff
@@ -0,0 +1,102 @@
+import { defaultClient } from "../client";
+import { ORDER_SIDE_ASK, ORDER_SIDE_BID, ORDER_TYPE_LIMIT, TestUser } from "../config";
+
+interface LiquidityBotConfig {
+  base: string;
+  quote: string;
+  deviation: number;
+  tiersAmount: number;
+  userId: TestUser;
+}
+
+class LiquidityBot {
+  market: string;
+  makerFee = 0;
+  takerFee = 0;
+  balanceCache = {};
+
+  constructor(private config: LiquidityBotConfig) {
+    this.market = `${config.base}_${config.quote}`;
+  }
+
+  async init() {
+    await defaultClient.connect();
+  }
+
+  async tick() {
+    const { quote, base, userId, deviation, tiersAmount } = this.config;
+    const balance = await defaultClient.balanceQuery(userId);
+
+    const balanceQuote = +balance.get(quote).available + +balance.get(quote).frozen;
+    const balanceBase = +balance.get(base).available + +balance.get(base).frozen;
+
+    if (this.balanceCache[quote] === balanceQuote && this.balanceCache[base] === balanceBase) {
+      console.log("not trades happend, skipping");
+      return;
+    }
+
+    this.balanceCache[quote] = balanceQuote;
+    this.balanceCache[base] = balanceBase;
+
+    const getPrice = (tier: number, side: number) => {
+      return (balanceQuote * Math.pow(deviation, side * (2 * tier + 1))) / balanceBase;
+    };
+
+    const orders = new Array(tiersAmount).fill(null).reduce((acc: any[], _, tier) => {
+      const askPrice = getPrice(tier, 1);
+      const askAmount = balanceBase / Math.pow(deviation, tier) - balanceBase / Math.pow(deviation, tier + 1);
+      const askOrder = this.createOrder(ORDER_SIDE_ASK, askPrice, askAmount);
+
+      const bidPrice = getPrice(tier, -1);
+      const bidAmount = balanceBase * Math.pow(deviation, tier + 1) - balanceBase * Math.pow(deviation, tier);
+      const bidOrder = this.createOrder(ORDER_SIDE_BID, bidPrice, bidAmount);
+
+      acc = [...acc, askOrder, bidOrder];
+      return acc;
+    }, []);
+
+    await defaultClient.batchOrderPut(userId, this.market, true, orders);
+  }
+
+  createOrder(order_side: number, price: number, amount: number) {
+    return {
+      market: this.market,
+      order_side,
+      order_type: ORDER_TYPE_LIMIT,
+      price: price.toFixed(6),
```

Comment:
As mentioned in the last review [here](https://github.com/BitRAW/dingir-exchange/pull/11#discussion_r790014124) and [here](https://github.com/BitRAW/dingir-exchange/pull/11#discussion_r790014501),  **BID** orders should always round the price **down** while **ASK** orders should always round the price **up**.

Correct me if I see this wrong. I think if a price becomes small (like 0.0000444) this could indeed become significant.